src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,86 +167,565 @@ def __repr__(self):
+
+
+# Feature Flag Helper Functions #
+def custom_serialize_conditions(conditions_dict):
+    """
+    Helper Function to serialize Conditions
+
+    Args:
+        conditions_dict - Dictionary of {str, Any}
+
+    Return:
+        JSON serializable Dictionary
+    """
     featurefilterdict = {}
 
     for key, value in conditions_dict.items():
         if key == FeatureFlagConstants.CLIENT_FILTERS:
-            featurefilterdict[key] = [feature_filter.__dict__ for feature_filter in value]
+            featurefilterdict[key] = [
+                feature_filter.__dict__ for feature_filter in value
+            ]
         else:
             featurefilterdict[key] = value
 
     return featurefilterdict
 
 
+def custom_serialize_allocation(allocation):
+    """
+    Helper Function to serialize Allocation
+
+    Args:
+        allocation_dict - FeatureAllocation object
+
+    Return:
+        JSON serializable Dictionary
+    """
+    featureallocationdict = {}
+
+    if allocation.user:
+        featureallocationdict[FeatureFlagConstants.USER] = [
+            user.__dict__ for user in allocation.user
+        ]
+
+    if allocation.group:
+        featureallocationdict[FeatureFlagConstants.GROUP] = [
+            group.__dict__ for group in allocation.group
+        ]
+
+    if allocation.percentile:
+        featureallocationdict[FeatureFlagConstants.PERCENTILE] = [
+            percentile.__dict__ for percentile in allocation.percentile
+        ]
+
+    if allocation.default_when_enabled:
+        featureallocationdict[FeatureFlagConstants.DEFAULT_WHEN_ENABLED] = allocation.default_when_enabled
+
+    if allocation.default_when_disabled:
+        featureallocationdict[FeatureFlagConstants.DEFAULT_WHEN_DISABLED] = allocation.default_when_disabled
+
+    if allocation.seed:
+        featureallocationdict[FeatureFlagConstants.SEED] = allocation.seed
+
+    return featureallocationdict
+
+
+def custom_serialize_variants(variants_list):
+    """
+    Helper Function to serialize Variants
+
+    Args:
+        variants_list - List of FeatureVariant objects
+
+    Return:
+        JSON serializable List
+    """
+    featurevariants = []
+    for variant in variants_list:
+        variant_dict = {}
+        variant_dict[FeatureFlagConstants.NAME] = variant.name
+        variant_dict[FeatureFlagConstants.VARIANT_CONFIGURATION_VALUE] = variant.configuration_value
+
+        if variant.status_override:
+            variant_dict[FeatureFlagConstants.VARIANT_STATUS_OVERRIDE] = variant.status_override
+
+        featurevariants.append(variant_dict)
+    return featurevariants
+
+
+def custom_serialize_telemetry(telemetry):
+    """
+    Helper Function to serialize Telemetry
+
+    Args:
+        telemetry - FeatureTelemetry object
+
+    Return:
+        JSON serializable dictionary
+
+    """
+
+    feature_telemetry_dict = {}
+
+    if telemetry.enabled:
+        feature_telemetry_dict[FeatureFlagConstants.TELEMETRY] = telemetry.enabled
```

Comment:
telemetry.enabled is boolean, we will not add the enabled property if it is false. Is this intended?